ui/hooks/useGasFeeEstimates.js

```diff
@@ -25,14 +32,44 @@ import { useSafeGasEstimatePolling } from './useSafeGasEstimatePolling';
  * GasFeeController that it is done requiring new gas estimates. Also checks
  * the returned gas estimate for validity on the current network.
  *
+ * @param _networkClientId - The optional network client ID to get gas fee estimates for. Defaults to the currently selected network.
  * @returns {GasEstimates} GasEstimates object
  */
-export function useGasFeeEstimates() {
-  const gasEstimateType = useSelector(getGasEstimateType);
-  const gasFeeEstimates = useSelector(getGasFeeEstimates, isEqual);
-  const isGasEstimatesLoading = useSelector(getIsGasEstimatesLoading);
-  const isNetworkBusy = useSelector(getIsNetworkBusy);
-  useSafeGasEstimatePolling();
+export function useGasFeeEstimates(_networkClientId) {
+  const selectedNetworkClientId = useSelector(getSelectedNetworkClientId);
+  const networkClientId = _networkClientId ?? selectedNetworkClientId;
+
+  const [chainId, setChainId] = useState('');
+
+  const gasEstimateType = useSelector((state) =>
+    getGasEstimateTypeByChainId(state, chainId),
+  );
+  const gasFeeEstimates = useSelector(
+    (state) => getGasFeeEstimatesByChainId(state, chainId),
+    isEqual,
+  );
+  const isGasEstimatesLoading = useSelector((state) =>
+    getIsGasEstimatesLoadingByChainId(state, { chainId, networkClientId }),
+  );
+  const isNetworkBusy = useSelector((state) =>
+    getIsNetworkBusyByChainId(state, chainId),
+  );
+
+  useEffect(() => {
+    getNetworkConfigurationByNetworkClientId(networkClientId).then(
+      (networkConfig) => {
+        if (networkConfig) {
```

Comment:
could this be undefined?

Reply:
doesn't seem like we have any plan in the case that its not... should we throw an error?

Reply:
If its `undefined` we could set some error state that we could return from the hook. The reason I think we probably need to do something like this is because it won't be apparent what's wrong and why polling isn't working if this failed... Though I admit I don't see why it would fail